importer/balanced/builder.go

```diff
@@ -27,87 +43,114 @@ import (
 // and depth only increases when the tree is full, that is, when
 // the root node has reached the maximum number of links.
```

Comment:
In keeping with the diagrams I suggested in other sections, this could help here:

```go
//            +-------------+        
//            |   Root 1    |        
//            +-------------+        
//                   |               
//      ( fillNodeRec fills in the )
//      ( chunks on the root.      )
//                   |
//            +------+------+        
//            |             |        
//       + - - - - +   + - - - - +
//       | Chunk 1 |   | Chunk 2 |
//       + - - - - +   + - - - - +
//
//                          ↓
//     When the root is full but there's more data...
//                          ↓
//
//            +-------------+        
//            |   Root 1    |        
//            +-------------+        
//                   |               
//            +------+------+        
//            |             |        
//       +=========+   +=========+   + - - - - +
//       | Chunk 1 |   | Chunk 2 |   | Chunk 3 |
//       +=========+   +=========+   + - - - - +
//
//                          ↓
//     ...Layou's job is to create a new root.
//                          ↓
//
//                           +-------------+
//                           |   Root 2    |
//                           +-------------+
//                                 |
//                   +-------------+ - - - - - - - - +
//                   |                               |
//            +-------------+            ( fillNodeRec creates the )
//            |   Root 1    |            ( branch that connects    )
//            +-------------+            ( "Root 2" to "Chunk 3."  )
//                   |                               |
//            +------+------+             + - - - - -+
//            |             |             |
//       +=========+   +=========+   + - - - - +
//       | Chunk 1 |   | Chunk 2 |   | Chunk 3 |
//       +=========+   +=========+   + - - - - +
```

Reply:
Yes, that's exactly the type of in-progress creation I was writing about above, great!